pkg/jobstore/inmemory/inmemory.go

```diff
@@ -146,25 +136,54 @@ func (d *InMemoryJobStore) GetJobs(ctx context.Context, query jobstore.JobQuery)
 
 	listSorter := func(i, j int) bool {
 		switch query.SortBy {
-		case "id":
+		case "modified_at":
 			if query.SortReverse {
-				// what does it mean to sort by ID?
-				return result[i].ID > result[j].ID
+				return result[i].ModifyTime > result[j].ModifyTime
 			} else {
-				return result[i].ID < result[j].ID
+				return result[i].ModifyTime < result[j].ModifyTime
 			}
-		case "created_at":
+		default:
+			// We apply createdat as a default sort so that we can use it for pagination.
+			// Without a known default we won't have a stable sort that makes sense for
+			// offsets/limits.
 			if query.SortReverse {
 				return result[i].CreateTime > result[j].CreateTime
 			} else {
 				return result[i].CreateTime < result[j].CreateTime
 			}
-		default:
-			return false
 		}
 	}
 	sort.Slice(result, listSorter)
-	return result, nil
+
+	if query.Selector != nil {
+		var filtered []models.Job
+		for _, j := range result {
+			if query.Selector.Matches(labels.Set(j.Labels)) {
+				filtered = append(filtered, j)
+			}
+		}
+		result = filtered
+	}
+
+	// To be able to extract based on offset and limit, we need the records to be sorted in
+	// the same order for each query.  This is why we sort the list above and why the offset
+	// and limit are applied so late.
+	if query.Offset > 0 && query.Offset <= uint32(len(result)) {
+		result = result[query.Offset:]
+	}
+
+	if query.Limit > 0 && query.Limit <= uint32(len(result)) {
+		result = result[:query.Limit+query.Offset]
+	}
```

Comment:
result already starts from the offset after line 172. I think you only need to do `result[:query.limit]`. Are we still using inmemory store anywhere? Should we get rid of it?

Reply:
Yeah, let's delete it in a new PR, boltdb has had enough time to settle in now, and if we're adding NATS KV soon then even this may not be necessary.